ios/brave-ios/Sources/Brave/Frontend/Browser/Toast.swift

```diff
@@ -97,6 +99,14 @@ class Toast: UIView {
   }
 
   @objc func handleTap(_ gestureRecognizer: UIGestureRecognizer) {
+    if tapDismissalMode == .outsideTap {
+      let location = gestureRecognizer.location(in: self)
+      // Check if the tap was inside the toast view
+      if self.point(inside: location, with: nil) {
+        return
+      }
+    }
```

Comment:
I dont think this will work, a gesture recognizer won't fire its events on a point outside of the view in the first place

Reply:
The gesture recognizer which handles this tap is added to the superview of the Toast. So taps can be both inside and outside of Toast:
```
  override func didMoveToSuperview() {
    super.didMoveToSuperview()
    superview?.addGestureRecognizer(gestureRecognizer)
  }
```

Reply:
adding a gesture recognizer to the superview like this doesn't seem like the right way to go. Is this just to dismiss it without having to tap the x? Maybe we'd be better dismissing it automatically after a timeout period?

Reply:
Actually all current Toasts uses gesture recognizer in the superview already. This was added to Toast class before the current PR. Here is a link in the current master branch: https://github.com/brave/brave-core/blob/master/ios/brave-ios/Sources/Brave/Frontend/Browser/Toast.swift#L31
The current PR just adds a flag for not dismissing a Toast when tapped inside the Toast itself.

Basing on comments in the spec, the Toast shouldn't be automatically dismissed by timeout.

Reply:
Alright thanks, maybe we'll clean this up later then